x-pack/plugins/cases/public/components/all_cases/severity_filter.tsx

```diff
@@ -5,61 +5,42 @@
  * 2.0.
  */
 
-import type { EuiSuperSelectOption } from '@elastic/eui';
-import { EuiFlexGroup, EuiFlexItem, EuiHealth, EuiSuperSelect, EuiText } from '@elastic/eui';
+import type { EuiSelectableOption } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiHealth } from '@elastic/eui';
 import React from 'react';
-import type { CaseSeverityWithAll } from '../../containers/types';
-import { SeverityAll } from '../../containers/types';
-import { severitiesWithAll } from '../severity/config';
+import type { CaseSeverity } from '../../../common/types/domain';
+import type { FilterOptions } from '../../containers/types';
+import { severities } from '../severity/config';
+import { MultiSelectFilter } from './multi_select_filter';
+import * as i18n from './translations';
 
 interface Props {
-  selectedSeverity: CaseSeverityWithAll;
-  onSeverityChange: (status: CaseSeverityWithAll) => void;
-  isLoading: boolean;
-  isDisabled: boolean;
+  selectedOptions: CaseSeverity[];
+  onChange: ({ filterId, options }: { filterId: keyof FilterOptions; options: string[] }) => void;
 }
 
-export const SeverityFilter: React.FC<Props> = ({
-  selectedSeverity,
-  onSeverityChange,
-  isLoading,
-  isDisabled,
-}) => {
-  const caseSeverities = Object.keys(severitiesWithAll) as CaseSeverityWithAll[];
-  const options: Array<EuiSuperSelectOption<CaseSeverityWithAll>> = caseSeverities.map(
-    (severity) => {
-      const severityData = severitiesWithAll[severity];
-      return {
-        value: severity,
-        inputDisplay: (
-          <EuiFlexGroup
-            gutterSize="xs"
-            alignItems={'center'}
-            responsive={false}
-            data-test-subj={`case-severity-filter-${severity}`}
-          >
-            <EuiFlexItem grow={false}>
-              {severity === SeverityAll ? (
-                <EuiText size="s">{severityData.label}</EuiText>
-              ) : (
-                <EuiHealth color={severityData.color}>{severityData.label}</EuiHealth>
-              )}
-            </EuiFlexItem>
-          </EuiFlexGroup>
-        ),
-      };
-    }
-  );
+const options = Object.keys(severities) as CaseSeverity[];
+
+export const SeverityFilter: React.FC<Props> = ({ selectedOptions, onChange }) => {
+  const renderOption = (option: EuiSelectableOption) => {
+    const severityData = severities[option.label as CaseSeverity];
```

Comment:
I think the component should be resilient in case the `option.label` is not a severity. We can fallback to no color if it does not exist.

Reply:
I'm not sure I'm following. At this point it will be a severity as the options available, which are the options that will be passed though this function, are imported in this file and it's a config created by us. This is gonna render the options in the selectable, it doesn't render the value set as severity for the case (if that's what you might have thought?)

Reply:
Sorry for the confusion. What I wanted to communicate is that are we sure that the `option.label` will always be a valid `CaseSeverity`? Should the component be protected against values that are not `CaseSeverity`?

Reply:
I'm pretty confident that it will be a case severity. I updated it to use the EuiSelectableOption Generic so now we override the type of label and don't need to cast it anymore. Same for status